screen-pop-app/screen-pop/src/notification.tsx

```diff
@@ -0,0 +1,176 @@
+import axios from "axios";
+import React, { useState, useEffect, useRef } from "react";
+import { ConfigData, ConversationManager } from "./ConversationManager";
+
+const ENVIRONMENT = "mypurecloud.com"
+const SCREENPOP_URL = "https://developer.genesys.cloud/";
+interface ConversationDetails {
+  token: string;
+  interactionID: string;
+  userID: string;
+  configData?: ConfigData; 
+}
+
+export const Channel = ({
+  token,
+  interactionID,
+  userID,
+  configData
+}: ConversationDetails) => {
+   // eslint-disable-next-line
+  const [chatStatusDisplay, setChatStatusDisplay] = useState<string>(""); //this property is to trigger new display of chatState 
+  const prevStatus = useRef(configData?.lastState || "");
+  const chatState = useRef(configData?.lastState || ""); 
+  useEffect(() => {
+    if(chatState.current === 'disconnected') return 
+    if (token) {
+      let channelExists: boolean = false; 
+
+      axios({ //get existing channels 
```

Comment:
Add a lot more comments in this file to explain what each step is doing. This is a core part of what this blueprint seeks to demonstrate, and verbose comments will help readers understand what's going on without having to compile the code in their heads.

Reply:
added some comments on last commit but let me know if I need more